src/Libraries/DSCPython/Encoders/ListEncoder.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Dynamo.Utilities;
+using Python.Runtime;
+
+namespace DSCPython.Encoders
+{
+    internal class ListEncoder : IPyObjectEncoder, IPyObjectDecoder
+    {
+        private static readonly Type[] decodableTypes = new Type[]
+        {
+            typeof(IList), typeof(ArrayList),
+            typeof(IList<>), typeof(List<>)
+        };
+
+        public bool CanDecode(PyObject objectType, Type targetType)
+        {
+            if (targetType.IsGenericType)
+            {
+                targetType = targetType.GetGenericTypeDefinition();
+            }
+            return decodableTypes.IndexOf(targetType) >= 0;
+        }
+
+        public bool CanEncode(Type type)
+        {
+            return typeof(IList).IsAssignableFrom(type);
+        }
+
+        public bool TryDecode<T>(PyObject pyObj, out T value)
+        {
+            if (!PySequence.IsSequenceType(pyObj))
+            {
+                value = default;
+                return false;
+            }
+
+            using (var pyList = PyList.AsList(pyObj))
+            {
+                if (typeof(T).IsGenericType)
+                {
+                    value = pyList.ToList<T>();
+                }
+                else
+                {
+                    value = (T)pyList.ToList();
+                }
+                return true;
+            }
+        }
+
+        public PyObject TryEncode(object value)
+        {
+            // This is a no-op to prevent Python.NET from encoding generic lists
+            // https://github.com/pythonnet/pythonnet/pull/963#issuecomment-642938541
+            return PyObject.FromManagedObject(value);
```

Comment:
does this break examples where I want to call a python function with a list of ints?

```
def sum(data):
    for i in data:
```
where data is a List<int> ?

Reply:
Good question. Let me try it and come back with the answer.

Reply:
It works. Do you want me to add it to the test?
<img width="1292" alt="Screen Shot 2020-07-08 at 12 39 42 PM" src="https://user-images.githubusercontent.com/10048120/86946509-77b5cb00-c118-11ea-98bb-b1eefb95a2b8.png">

Reply:
sure, nothing wrong with more tests?

Reply:
okay, so l2 is a .net list because Flatten is a zt function?

Reply:
Yes, it's `List.Flatten` from our core nodes